Lib/test/test_typing.py

```diff
@@ -4963,6 +4965,8 @@ def test_special_attrs(self):
             typing.Concatenate[Any, SpecialAttrsP]: 'Concatenate',
             typing.Final[Any]: 'Final',
             typing.Literal[Any]: 'Literal',
+            typing.Literal[1, 2]: 'Literal',
+            typing.Literal[True, 2]: 'Literal',
```

Comment:
Not sure if these tests for anything useful? It should still pass even without the fix.

Reply:
Ah, it passes without the fix because due to the bug `typing.Literal[True, 2]` and `typing.Literal[1, 2]` are the same.

But it fails with the PR originally proposed by @sobolevn. Some tests are added to catch possible similar errors.